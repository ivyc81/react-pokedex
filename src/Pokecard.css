.Pokecard {
    border-radius: 8px;
    background-color: #b5d5d4;
    width: 140px;
    text-align: center;
}