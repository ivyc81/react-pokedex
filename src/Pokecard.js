import React, { Component } from 'react';
import './Pokecard.css';

const IMG_URL = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/';

class Pokecard extends Component {
    render() {
        return (
            <div className='Pokecard'>
                <p>{ this.props.name }</p>
                <img src={ IMG_URL + this.props.id + ".png" } alt={ this.props.name } /><br/>
                <small>Type: { this.props.type }</small><br/>
                <small>EXP: { this.props.base_experience }</small>
            </div>
        );
    }
}

export default Pokecard;