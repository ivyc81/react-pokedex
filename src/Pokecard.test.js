import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Pokecard from "./Pokecard";

// smoke test
it("renders without crashing", function() {
  shallow(<Pokecard />);
});

// snapshot test
it("matches snapshot", function() {
  let wrapper = shallow(<Pokecard />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
