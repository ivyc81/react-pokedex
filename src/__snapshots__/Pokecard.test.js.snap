// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <p />
  <img
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
  />
  <small>
    Type: 
  </small>
  <small>
    EXP: 
  </small>
</div>
`;
